### PR TITLE
add Initialization docs

### PR DESCRIPTION
**Pull request by balazsorban44**
Based on a few questions, this documentation could be the source of truth when explaining how to do custom actions that are too specific to be included in the library.

Some related discussions:
https://github.com/nextauthjs/next-auth/discussions/3133
https://github.com/nextauthjs/next-auth/discussions/3156
https://github.com/nextauthjs/next-auth/discussions/2703

### CODE REVIEW

File: docs/configuration/initialization.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  })
+}
+```
+
+To completely understand what actions correspond to and what methods are supported, check out the [REST API documentation](/getting-started/rest-api), or look at the [source code](https://github.com/nextauthjs/next-auth/blob/beta/src/core/index.ts).
+
+This way of initializing `NextAuth` is very powerful, but should be used sparingly.
+
+:::warning
+Changing parts of the request that is essential to `NextAuth` to do it's job - like messing with the [default cookies](http://localhost:3000/configuration/options#cookies) - can have unforeseen consequences, and have the potential to introduce security holes if done incorrectly. Only change those this if you understand these consequences.
```
Comment (zakaryan2004) [inline docs/configuration/initialization.md:128]:
```suggestion
Changing parts of the request that is essential to `NextAuth` to do it's job - like messing with the [default cookies](/configuration/options#cookies) - can have unforeseen consequences, and have the potential to introduce security holes if done incorrectly. Only change those this if you understand these consequences.
```

Comment (zakaryan2004) [inline docs/configuration/initialization.md:128]:
```suggestion
Changing parts of the request that is essential to `NextAuth` to do it's job - like messing with the [default cookies](/configuration/options#cookies) - can have unforeseen consequences, and have the potential to introduce security holes if done incorrectly. Only change those this if you understand these consequences.
```

**Review by zakaryan2004 — CHANGES_REQUESTED** [review]
Great idea! This will help a lot of users.

I'll test the provider idea again in my project and see if it works this way.

File: docs/configuration/initialization.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import NextAuth from "next-auth"
+
+export default NextAuth({
+  ...
+})
+```
+
+Here, you only need to pass your [options](/configuration/options) to `NextAuth`, and `NextAuth` does the rest.
+
+This is the preferred initialization in tutorials/other parts of the documentation, as it somewhat simplifies the code and there is smaller chances for errors in the authentication flow.
```
Comment (ndom91) [inline docs/configuration/initialization.md:26]: How about to simplify this sentence, "..as it simplifies the code and reduces potential errors in the authentication flow" ?

File: docs/configuration/initialization.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+## Advanced initialization
+
+In rare cases, there is just **some extra functionality** that you wish NextAuth.js had, or you need to _preprocess_ the request, but you understand that it is way too specific for your use-case to be introduced as part of the library.
```
Comment (ndom91) [inline docs/configuration/initialization.md:30]: I would tighten up this intro part a bit, maybe something along the lines of - "If you have a specific use case and need to make NextAuth.js do something slightly different than what its designed for, keep in mind, the `[...nextauth].js` config file is still just **a regular [API Route]()** at the end of the day.."

File: docs/configuration/initialization.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  if (isDefaultSigninPage) providers.pop()
+
+  return await NextAuth(req, res, {
+    providers,
+    ...
+  })
+}
+```
+
+To completely understand what actions correspond to and what methods are supported, check out the [REST API documentation](/getting-started/rest-api), or look at the [source code](https://github.com/nextauthjs/next-auth/blob/beta/src/core/index.ts).
```
Comment (ndom91) [inline docs/configuration/initialization.md:123]: Maybe simplify this as well, something along the lines of - "For more details on all available actions and which methods are supported, please check out the [REST API Documentation]() or the appropriate area in [the source code]()".

File: docs/configuration/initialization.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+This way of initializing `NextAuth` is very powerful, but should be used sparingly.
+
+:::warning
+Changing parts of the request that is essential to `NextAuth` to do it's job - like messing with the [default cookies](/configuration/options#cookies) - can have unforeseen consequences, and have the potential to introduce security holes if done incorrectly. Only change those this if you understand these consequences.
```
Comment (ndom91) [inline docs/configuration/initialization.md:128]: Good idea with the warning at the end here! "This method of initializing NextAuth.js is very powerful, so if you fuck it up its on you ;)" :joy:

**Review by ndom91 — APPROVED** [review]
Looks mostly good to me! I'm just gonna approve it, but I would either now or sometime later 
before going live with v4 GA tighten up some of the copy. Some parts are a bit wordy and could be slightly confusing. 

Great idea with this page though, its a super frequently asked about thing huh.
